sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/HeadingHasAccessibleContentCheck.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * SonarSource HTML analyzer :: Sonar Plugin
+ * Copyright (c) 2010-2024 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.checks.accessibility;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.api.Helpers;
+import org.sonar.plugins.html.api.BufferStack;
+import org.sonar.plugins.html.api.HtmlConstants;
+import org.sonar.plugins.html.checks.AbstractPageCheck;
+import org.sonar.plugins.html.node.DirectiveNode;
+import org.sonar.plugins.html.node.ExpressionNode;
+import org.sonar.plugins.html.node.TagNode;
+import org.sonar.plugins.html.node.TextNode;
+import org.sonar.plugins.html.node.Attribute;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.List;
+
+@Rule(key = "S6850")
+public class HeadingHasAccessibleContentCheck extends AbstractPageCheck {
+  private final List<String> invalidAttributes = List.of(
+    "aria-hidden"
+  );
+
+  private final List<String> vueJsContentLikeAttributes = List.of(
+    "v-html",
+    "v-text"
+  );
+
+  private final BufferStack bufferStack = new BufferStack();
+
+  private final Deque<TagNode> openingHeadingTags = new ArrayDeque<>();
+
+  @Override
+  public void startElement(TagNode node) {
+    if (Helpers.isHeadingTag(node)) {
+      bufferStack.start();
+      openingHeadingTags.push(node);
+
+      if (hasAnInvalidAttribute(node)) {
+        createViolation(node);
+      }
+    } else {
+      String nodeName = node.getNodeName();
+
+      // tags that are not part of the known HTML tags list are considered as content
+      if (!HtmlConstants.KNOWN_HTML_TAGS.contains(nodeName)) {
+        bufferStack.write(nodeName);
+      }
+    }
+
+    // vueJS attributes that maps to content are considered as content
+    vueJsContentLikeAttributes.forEach(attributeName -> {
+      String nodeAttribute = node.getAttribute(attributeName);
+
+      if (nodeAttribute != null && !nodeAttribute.isBlank() && bufferStack.getLevel() > 0) {
+        bufferStack.write(nodeAttribute);
+      }
+    });
+
+    super.startElement(node);
```

Comment:
do we need to call super?

Reply:
I don't know. Is there a convention with Java - or at SonarSource - that allows us to know if we must call a method parent's implementation?

Reply:
It's not really about convention but rather about intention and needs:

- In a constructor, you _must_ call the parent constructor to first initialize its state before proceeding with the child's. 
- In a method, you _could_ call the parent's implementation of the method if you need to trigger a side-effect you need.

I don't think it makes sense to make these calls, especially when the parent does nothing:

https://github.com/SonarSource/sonar-html/blob/4de7dfe5ceb1e7964f7a31e2ff8270a5b0bae8c5/sonar-html-plugin/src/main/java/org/sonar/plugins/html/visitor/DefaultNodeVisitor.java#L68-L70

Reply:
Yes, I will remove them because, here, we _know_ the parent's implementations do nothing.

> In a method, you could call the parent's implementation of the method if you need to trigger a side-effect you need.

Can I assume that, if the parent's implementation has side-effect (or _must_ be called), it would be documented in the parent class?

Reply:
Mmm, well if it's the case and we have to, sure, but I would first find the author of the parent's class, send them to Mordor, throw them to Mount Doom, and then, if they are still alive give a 1-hour lecture about inheritance, polymorphism, and encapsulation.